lms/djangoapps/courseware/management/commands/fix_student_module_newlines.py

```diff
@@ -0,0 +1,212 @@
+"""
+Fix StudentModule entries that have Course IDs with trailing newlines.
+
+Due to a bug, many rows in courseware_studentmodule were written with a
+course_id that had a trailing newline. This command tries to fix that, and to
+merge that data with data that might have been written to the correct course_id.
+"""
+from collections import namedtuple
+from optparse import make_option
+from textwrap import dedent
+import logging
+
+from django.db import DatabaseError
+from django.core.management.base import BaseCommand, CommandError
+
+from courseware.models import StudentModule, BaseStudentModuleHistory
+from util.query import use_read_replica_if_available
+
+log = logging.getLogger("fix_student_module_newlines")
+
+FixResult = namedtuple('FixResult', 'record_trimmed, data_copied, record_deleted, error')
+
+
+class Command(BaseCommand):
+    """Fix StudentModule entries that have Course IDs with trailing newlines."""
+    args = "<start_date> <end_date>"
+    help = dedent(__doc__).strip()
+    option_list = BaseCommand.option_list + (
+        make_option('--dry_run',
+                    action='store_true',
+                    default=False,
+                    help="Run read querys and say what we're going to do, but don't actually do it."),
+    )
+
+    def handle(self, *args, **options):
+        """Fix newline courses in CSM!"""
+        if len(args) != 2:
+            raise CommandError('Must specify start and end dates: e.g. "2016-08-23 16:43:00" "2016-08-24 22:00:00"')
+
+        start, end = args
+        dry_run = options['dry_run']
+
+        log.info(
+            "Starting fix_student_module_newlines in %s mode!",
+            "dry_run" if dry_run else "real"
+        )
+
+        rows_to_fix = use_read_replica_if_available(
+            StudentModule.objects.raw(
+                "select * from courseware_studentmodule where modified between %s and %s and course_id like %s",
```

Comment:
Because I couldn't figure out a way to escape % hard enough -- the query parser kept on wanting to read it as something to substitute for, regardless of whether I put a backslash in front of it or not. This bit of code does work. It's being parsed by a layer that understands how to do the substitution and protect against injection attacks.